tests/unit/tracerTest.cpp

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <tilck/common/basic_defs.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+#include <tilck/common/string_util.h>
+#include <tilck/common/printk.h>
+
+#include <tilck/kernel/user.h>
+#include <tilck/mods/tracing.h>
+
+#include <tilck/kernel/test/tracing.h>
+}
+
+class MockingTracer : public KernelSingleton {
+public:
+
+   MOCK_METHOD(int, copy_str_from_user, (void *, const void *, size_t, size_t *), (override));
+   MOCK_METHOD(int, copy_from_user, (void *, const void *, size_t), (override));
+};
+
+TEST(tracer_test, save_param_buffer)
+{
+   MockingTracer mock;
+
+   long data_sz = -1;
+   void *data = (void *)"test";
+
+   char dest_buf_1[8];
+   size_t dest_bs_1 = sizeof(dest_buf_1);
+
+   char dest_buf_2[8];
+   size_t dest_bs_2 = sizeof(dest_buf_2);
+
+   char dest_buf_3[8];
+   size_t dest_bs_3 = sizeof(dest_buf_3);
+
+   EXPECT_CALL(mock, copy_str_from_user)
+      .WillOnce(Return(-1))
+      .WillOnce(Return(1));
+
+   // rc < 0
+   EXPECT_TRUE(save_param_buffer(data, data_sz, dest_buf_1, dest_bs_1));
+   EXPECT_STREQ(dest_buf_1, "<fault>");
+
+   // rc > 0
+   EXPECT_TRUE(save_param_buffer(data, data_sz, dest_buf_2, dest_bs_2));
+   EXPECT_EQ(dest_buf_2[dest_bs_2 - 1], 0);
```

Comment:
It's not really enough to test this case this way. See the code in `save_param_buffer()`:
```C
static bool
save_param_buffer(void *data, long data_sz, char *dest_buf, size_t dest_bs)
{
   if (data_sz == -1) {

      /* assume that `data` is a C string */
      int rc = copy_str_from_user(dest_buf, data, dest_bs, NULL);

      if (rc < 0) {

         /* Reading from `data` caused a PAGE fault */
         memcpy(dest_buf, "<fault>", 8);

      } else if (rc > 0) {

         /*
          * The user buffer is bigger than our reserved space: just truncate it.
          */
         dest_buf[dest_bs - 1] = 0;
      }
```
if `copy_str_from_user()` returned > 0, we expect the dest buffer to be truncated, but it needs to also have some content. The current test just checks that the dest buffer has been truncated. To make it really work, in one of the cases copy_str_from_user() should have an actual implementation as strcpy(), instead of just returning 1.

To do that, see "Using Functions/Methods/Functors/Lambdas as Actions" in http://google.github.io/googletest/gmock_cook_book.html. Briefly, you can pass a lambda to WillOnce, or use Invoke() and then pass a class + method. 

In alternative, if that's end up taking too much time, at least initialize the dest buffer with something well known and then check that the content it's the same but truncated.